src/backend/chat/custom/utils.py

```diff
@@ -16,22 +14,11 @@ def get_deployment(name: str, ctx: Context, **kwargs: Any) -> BaseDeployment:
 
     Returns:
         BaseDeployment: Deployment implementation instance based on the deployment name.
-
-    Raises:
-        ValueError: If the deployment is not supported.
     """
     kwargs["ctx"] = ctx
-    deployment = AVAILABLE_MODEL_DEPLOYMENTS.get(name)
-
-    # Check provided deployment against config const
-    if deployment is not None:
-        return deployment.deployment_class(**kwargs, **deployment.kwargs)
-
-    # Fallback to first available deployment
-    default = get_default_deployment(**kwargs)
-    if default is not None:
-        return default
+    try:
+        deployment = deployment_service.get_deployment_by_name(name)
```

Comment:
I don't think I understand what you're seeing. With the new code here, `deployment_service` will throw if it can't find a deployment with the specified name. In that case, we catch and instead return a default deployment. And if there are no available deployments at all, `get_default_deployment` will also throw.

Reply:
Ah I see, I confused the `get_deployment_by_name` call with a similarly named method I think. Good to go then.